### PR TITLE
compiles and runs successfully

### PR DESCRIPTION
**Pull request by maryamariyan**

### CODE REVIEW

File: src/System.ComponentModel.Composition/src/Strings.resx [hunk 1]
```diff
@@ -0,0 +1,153 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (danmoseley) [inline src/System.ComponentModel.Composition/src/Strings.resx]: We should not check in any strings.designer.cs files

Generally there should be exactly one strings.resx file and it should be in src/Resources, not here. Please check...

Comment (danmoseley) [reply src/System.ComponentModel.Composition/src/Strings.resx]: Also neither strings.resx nor strings.designer.cs should be in the project file at all. strings.resx is automatically found in the resources folder

File: src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/CompositionContractMismatchException.cs [hunk 2]
```diff
@@ -56,6 +57,16 @@ public CompositionContractMismatchException(string message, Exception innerExcep
...
+        /// Initializes a new instance of the <see cref="CompositionContractMismatchException"/> class with serialized data.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo"/> that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">The <see cref="StreamingContext"/>  that contains contextual information about the source or destination.</param>
+        protected CompositionContractMismatchException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+        }
+
 #if FEATURE_SERIALIZATION
```
Comment (danmoseley) [inline src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/CompositionContractMismatchException.cs]: Remove FEATURE_SERIALIZATION symbol everywhere.
Retain the content if it is in an exception type or tests for one. Delete it if it is not in an exception type.

File: src/System.ComponentModel.Composition/src/Microsoft/Internal/Assumes.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (danmoseley) [inline src/System.ComponentModel.Composition/src/Microsoft/Internal/Assumes.cs]: Delete this and use C:\git\corefx\src\Common\src\Microsoft\Internal\Assumes.cs

File: src/System.ComponentModel.Composition/tests/System/ComponentModel/IO/FileIO.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (danmoseley) [inline src/System.ComponentModel.Composition/tests/System/ComponentModel/IO/FileIO.cs]: these files in /IO/*cs duplicate the existing test helpers for creating and using temp files/directories.
leave as is for now but in future someone could eliminate the duplication

Comment (danmoseley) [reply src/System.ComponentModel.Composition/tests/System/ComponentModel/IO/FileIO.cs]: You have checked in 2 copies of them all though. You only need 1 copy. Please check for dupe C# files

File: src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/TypeCatalogTests.cs [hunk 5]
```diff
@@ -0,0 +1,574 @@
...
+
+// This is a glorious do nothing ReflectionContext
+    public class TypeCatalogTestsReflectionContext : ReflectionContext
+    {
+        public override Assembly MapAssembly(Assembly assembly)
+        {
+            return assembly;
+        }
+
+#if FEATURE_INTERNAL_REFLECTIONCONTEXT
```
Comment (danmoseley) [inline src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/TypeCatalogTests.cs]: We don't define any defines, generally. So for each one you must remove it, and make a decision whether to keep the code inside it. If they existed in the implementation, do the same thing we did for that. There shold probably be zero `#if` in the code.

File: src/System.ComponentModel.Composition/tests/IExportableTest.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+using System;
```
Comment (danmoseley) [inline src/System.ComponentModel.Composition/tests/IExportableTest.cs]: Many of your files are missing the license. To check, do something liek this:
`for /f %i in ('dir /s/b *') do @(@echo %i & head -4 %i)`

I got `head` etc from the GNU tools for Win32 or you can use bash.

Always look for ways to get a simple script to do the hard work for you :)

File: src/System.ComponentModel.Composition/src/Microsoft/Internal/Assumes.cs (hunk 3 above)
Comment (danmoseley) [reply src/System.ComponentModel.Composition/src/Microsoft/Internal/Assumes.cs]: and other assumes file

**Comment by danmoseley** [comment]
You can make a PR..

File: src/System.ComponentModel.Composition/tests/IExportableTest.cs (hunk 6 above)
Comment (maryamariyan) [reply src/System.ComponentModel.Composition/tests/IExportableTest.cs]: I'll make sure to address this as well. thanks

**Pull request closed without merging**